common/init.sls

```diff
@@ -1,11 +1,38 @@
 {% from tpldir ~ '/map.jinja' import config with context %}
 
 {% if grains['kernel'] != 'Darwin' %}
-python-pkgs:
+# Ubuntu has python2 as default python
+python2:
   pkg.installed:
     - pkgs:
-      - python-pip
+      - python
       - python-dev
+
+python3:
+  pkg.installed:
+    - pkgs:
+      - python3
+
+# Ensure pip is default by purging pip3
+pip:
+  pkg.installed:
+    - pkgs:
+      - python-pip
+    - reload_modules: True
+
+pip3:
+  pkg.purged:
+    - pkgs:
+      - python3-pip
+
+# Virtualenv package creates virtualenv and virtualenv-3.4 executables
```

Comment:
Okay, sounds good. So when you say `python: python3` in the homu setup below, does that mean that it will also use `virtualenv3`? Because homu doesn't even install properly with an older virtualenv version, it needs to use pip3, python3, and virtualenv3. (Not sure if virtualenv2 even lets you create a python3 env, but it's good to be certain)